File: dongyimai_sellergoods_interface/src/main/java/com/offcn/service/BrandService.java
package com.offcn.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    public List<TbBrand> findAll();

    public PageResult findByPage(int pageNum,int pageSize);

    public void add(TbBrand brand);

    public void update(TbBrand brand);

    public TbBrand findById(Long id);

    public void delete(Long[] ids);

    public PageResult findBySelectPage(TbBrand brand,int pageNum,int pageSize);

    public List<Map> selectBrandList();
}

File: dongyimai_manager_web/src/main/webapp/js/service/brandService.js
// 专门用来处理请求的
app.service("brandService",function($http){

    this.findAll = function(){
        return $http.get("../brand/findAll.do");
    }

    this.findByPage = function(pageNum,pageSize){
        return $http.get("../brand/findByPage.do?pageNum="+pageNum+"&pageSize="+pageSize)
    }

    this.update = function(entity){
        return $http.post("../brand/update.do",entity);
    }

    this.add = function(entity){
        return $http.post("../brand/add.do",entity)
    }

    this.findById = function(id){
        return $http.get("../brand/findById.do?id=" + id);
    }

    this.delete = function(ids){
        return $http.get("../brand/delete.do?ids=" +  ids);
    }
    // $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,$scope.searchEntity)
    this.search = function(pageNum,pageSize,entity){
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,entity);

    }

    this.selectBrandList = function(){
        return $http.get("../brand/selectBrandList.do");
    }
})